ecs/pod_definition_manager.go

```diff
@@ -147,6 +147,24 @@ func (m *BasicPodDefinitionManager) CreatePodDefinition(ctx context.Context, opt
 	return &item, nil
 }
 
-func (m *BasicPodDefinitionManager) shouldCache() bool {
+// DeletePodDefinition deletes a pod definition and deletes it from the cache if
+// it is using a cache.
+func (m *BasicPodDefinitionManager) DeletePodDefinition(ctx context.Context, id string) error {
+	if _, err := m.client.DeregisterTaskDefinition(ctx, &ecs.DeregisterTaskDefinitionInput{
+		TaskDefinition: aws.String(id),
+	}); err != nil {
+		return errors.Wrapf(err, "deregistering task definition '%s'", id)
+	}
+
+	if m.usesCache() {
+		if err := m.cache.Delete(ctx, id); err != nil {
```

Comment:
I believe you can do a one-line return here and always return what `err` is.

Reply:
Sure, I simplified this down.